File: JavaScript/Medium.js
import Tyres from "./Tyres.js";

// Esta clase es para las llantas medias (Medium)
export default class Medium extends Tyres {
  // Constructor que delega en Tyres
  constructor(A, B, C, D, lapsMax, isWet, tyreColor) {
    super(A, B, C, D, lapsMax, isWet, tyreColor);
  }

  // Si supera la durabilidad máxima, entra a pits
  Durability(laps) {
    if (laps > this.durabilityMax) {
      this.Pits();
      return (
        `the tyres have been on track ${laps} laps\n` +
        `The Mediums are being changed`
      );
    }
    return `the tyres have been on track ${laps} laps\nCan continue`;
  }

  // Presión recomendada para Medium
  PSI() {
    console.log("The air pressure required is 40");
  }
}

File: Go/Medium.go
package main
import "fmt"

type Medium struct{
	Tyres
}

func NewMedium (A, B, C, D string, lapsMax int, isWet bool, tyreColor string) Medium{
	tyres := NewTyres(A, B, C, D, lapsMax, isWet, tyreColor)
	return Medium{Tyres: tyres}
}

func (m Medium) Durability(laps int) string {
	if (laps > m.durabilityMax){
		m.Pits()
		return fmt.Sprintf("The tyres have been on track %d laps\nThe Mediums are being changed", laps)
	} else{
		return fmt.Sprintf("The tyres have been on track %d laps\nCan continue", laps)
	}
}

func (m Medium) PSI(){
	fmt.Println("The air pressure required is 40")
}